core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -575,7 +585,8 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
     if (advertisedListenersProp != null) {
       CoreUtils.listenerListToEndPoints(advertisedListenersProp, effectiveListenerSecurityProtocolMap, requireDistinctPorts=false)
     } else {
-      listeners
+      // Only use implicit broker listeners here. Implicit controller listeners are handled in effectiveAdvertisedControllerListeners.
+      listeners.filterNot(l => controllerListenerNames.contains(l.listenerName.value()))
```

Comment:
Do you need this? `advertisedListeners` is a private method that contains all of the advertise listeners if it is set or listeners if it is not set. `effectiveAdvertisedBrokerListeners` and `effectiveAdvertisedControllerListeners` do the necessary filtering.

Reply:
Okay. It looks like the `advertisedListeners` method is not as useful as I originally thought when I implemented it. If you agree, let's just remove the method and inline the different implementations in `effectiveAdvertisedBrokerListeners` and `effectiveAdvertisedControllerListeners`. What do you think?

Reply:
Yes, removed `advertisedListeners`. Thanks.